collection_json.py

```diff
@@ -6,6 +6,59 @@
 __version__ = '0.1.0'
 
 
+class ArrayProperty(object):
+
+    """Property which converts from any enumerable to a typed Array instance."""
+
+    def __init__(self, cls, name):
+        """Constructs typed array property
+
+        :param cls type: the type of objects expected in the array
+        :param name str: name of the property
+        """
+        self.cls = cls
+        self.name = name
+        self.value_attr = "_" + name
+
+    def __get__(self, instance, owner):
+        return getattr(instance, self.value_attr, None)
```

Comment:
According to the descriptor docs (https://docs.python.org/3/reference/datamodel.html?highlight=descriptor#object.__get__) this should raise an AttributeError if the attribute value can not be computed.